src/messaging/ExchangeContext.cpp

```diff
@@ -113,7 +113,7 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
     // If session requires MRP, NoAutoRequestAck send flag is not specified and is not a group exchange context, request reliable
     // transmission.
     bool reliableTransmissionRequested =
-        GetSessionHandle()->RequireMRP() && !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck) && !IsGroupExchangeContext();
+        GetSessionHandle()->AllowsMRP() && !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck) && !IsGroupExchangeContext();
```

Comment:
The IsGroupExchangeContext() check here is redundant, right?  Pre-existing, so maybe just file a followup issue.

Reply:
Yes, It is redundant. I didn't realize that earlier given how the checks are structured. Will leave them here for now but will file an issue.